### PR TITLE
Add edit credential types

### PR DESCRIPTION
**Pull request by nixocio**
Add feature to edit credential types.

See: https://github.com/ansible/awx/issues/7326

credential_type_edit

### CODE REVIEW

**Comment by nixocio** [comment]
~There is one extra case that I need to verify - changing this PR to WIP.~ Done. Ready to be reviewed.

File: awx/ui_next/src/screens/CredentialType/CredentialType.jsx [hunk 1]
```diff
@@ -65,17 +65,12 @@ function CredentialType({ i18n, setBreadcrumb }) {
...
-    cardHeader = null;
-  }
-
   if (!isLoading && contentError) {
     return (
       <PageSection>
         <Card>
           <ContentError error={contentError}>
-            {contentError.response?.status === 404 && (
+            {contentError?.response?.status === 404 && (
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/CredentialType/CredentialType.jsx]: I'm not sure that `contentError?.` is necessary.  It is checking if there is a `contentError`, which should be the case if it get past line 68

File: awx/ui_next/src/screens/CredentialType/CredentialTypeEdit/CredentialTypeEdit.jsx [hunk 2]
```diff
@@ -1,10 +1,45 @@
...
+    } catch (error) {
+      setSubmitError(error);
+    }
+  };
+
+  const handleCancel = () => {
+    history.push(detailsUrl);
+  };
   return (
     <PageSection>
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/CredentialType/CredentialTypeEdit/CredentialTypeEdit.jsx]: Screen Shot 2020-07-06 at 10 50 18 AM
You have an extra `<PageSection>`

File: awx/ui_next/src/components/FormField/FormSubmitError.jsx [hunk 3]
```diff
@@ -22,6 +22,8 @@ function FormSubmitError({ error }) {
       Object.values(error.response.data).forEach(value => {
         if (Array.isArray(value)) {
           messages = messages.concat(value);
+        } else {
```
Comment (AlexSCorey) [inline awx/ui_next/src/components/FormField/FormSubmitError.jsx:25]: We have a util, which isn't in a util file, called getErrorMessage, which I think can handle all of these cases of error messages.  https://github.com/ansible/awx/blob/devel/awx/ui_next/src/components/ErrorDetail/getErrorMessage.js.  I thin we should, at some point, begin to migrate over to that file and let that handle our error messages.  If you want to investigate it, please feel free, but it isn't a blocker to merge this.

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
This looks really good so far.  Fix that 1 page section issue and if you want to touch base about the error handling ping me.

File: awx/ui_next/src/components/FormField/FormSubmitError.jsx (hunk 3 above)
Comment (nixocio) [reply awx/ui_next/src/components/FormField/FormSubmitError.jsx:25]: Thanks for your suggestion. I looked in how we are handling this in other components. 

@keithjgrant filed https://github.com/ansible/awx/issues/7515, once this issue is sorted out, we will have a better pattern in how to deal those cases.

File: awx/ui_next/src/screens/CredentialType/CredentialTypeDetails/CredentialTypeDetails.jsx [hunk 4]
```diff
@@ -70,7 +70,7 @@ function CredentialTypeDetails({ credentialType, i18n }) {
             <Button
               aria-label={i18n._(t`edit`)}
               component={Link}
-              to={`credential_types/${id}/edit`}
+              to={`/credential_types/${id}/edit`}
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/CredentialType/CredentialTypeDetails/CredentialTypeDetails.jsx:73]: Good catch there

**Review by AlexSCorey — APPROVED** [review]
This looks good to me.  Well done!

**Comment by dsesami** [comment]
Two notes:
1) this suffers from the same API char limit bug in https://github.com/ansible/awx/issues/7581. That does not block this PR, it is an API issue.
2) Can form validation on this block move the `!` bubble to the left side, like it is in most IDEs? it looks strange on the right side.
Screenshot from 2020-07-08 15-45-45

**Review by dsesami — APPROVED** [review]
approving as the above validation is related to the overall component, and not this PR.

**Pull request merged**